Add popup and "Always Show Clickables" setting

sam-dixon opened the pull request:
- Added a popup page to the extension
- When "Always show clickables" is changed, the popup updates the setting with `chrome.storage.sync.set` and sends a message to the content script to update the settings in the injected script (injected scripts do not have access to `chrome.storage.sync`)
- Popup loads state of slider with `restoreSettings`
- New `DOMContentLoaded` listener in `content-script.ts` updates injected script settings so the setting is maintained across tabs
- Every editor state request checks if `settings.alwaysShowClickables` is true and shows/clears clickable overlays
<img width="318" alt="Screen Shot 2022-02-14 at 2 59 32 PM" src="https://user-images.githubusercontent.com/12766907/153961001-1b603e5e-9e59-419b-8cc4-436bd6eaaa49.png">

sam-dixon committed:
add popup page and match styling to client

sam-dixon committed:
implement message passing between popup and background, store settings

sam-dixon committed:
first functional prototype

sam-dixon committed:
add missing await in message passing from content to injected script, don't return immediately if always show clickables is true

sam-dixon committed:
Merge branch 'refactor' of github.com:serenadeai/chrome into settings

tmacwill commented on src/popup.html:
```diff
@@ -0,0 +1,18 @@
+      <input id="showClickables" type="checkbox" />
+      <span class="slider round"></span>
+    </label>
+    Always show clickables
```
super minor thing—let's call this something like "link overlays" in the UI, since "clickable" is a bit confusing.

tmacwill reviewed (dismissed):
super small request inline. I forget, did we already include `input, textarea` in the selector for links? would make this feature a bit more usable if we did, and I think that's just a one-line change.

thanks!

sam-dixon committed:
add show inputs and links, change UI phrasing

sam-dixon commented:
Changed the selector to show inputs and links and updated the phrasing in the UI.

I also noticed that our `inViewport` helper function had a quirk where the overlay boxes themselves could prevent some inputs from being registered as in the viewport. I just removed this heuristic for now.

sam-dixon committed:
remove debug prints

tmacwill reviewed (approved):
awesome, thanks!
